### ISSUE TITLE
x64-dynamic-linux build problem for geos (Geometry Engine Open Source)

### ISSUE THREAD

**Issue by SergeiDelov** [microsoft/vcpkg#16602]
When attempted to build Linux packages as dynamic libraries (see https://github.com/microsoft/vcpkg/blob/master/docs/examples/overlay-triplets-linux-dynamic.md) got the following log:
````
Building package geos[core]:x64-dynamic-linux...
-- [OVERLAY] Loading triplet configuration from: /mnt/c/vcpkg/custom-triplets/x64-dynamic-linux.cmake
-- Downloading http://download.osgeo.org/geos/geos-3.9.0.tar.bz2 -> geos-3.9.0.tar.bz2...
-- Extracting source /mnt/c/vcpkg/downloads/geos-3.9.0.tar.bz2
-- Applying patch dont-build-docs.patch
-- Applying patch dont-build-astyle.patch
-- Using source at /mnt/c/vcpkg/buildtrees/geos/src/3.9.0-27800c28d2.clean
-- Configuring x64-dynamic-linux-dbg
-- Configuring x64-dynamic-linux-rel
-- Building x64-dynamic-linux-dbg
-- Building x64-dynamic-linux-rel
-- Performing post-build validation
There should be no empty directories in /mnt/c/vcpkg/packages/geos_x64-dynamic-linux
The following empty directories were found:

    /mnt/c/vcpkg/packages/geos_x64-dynamic-linux/bin
    /mnt/c/vcpkg/packages/geos_x64-dynamic-linux/debug/bin

If a directory should be populated but is not, this might indicate an error in the portfile.
If the directories are not needed and their creation cannot be disabled, use something like this in the portfile to remove them:

    file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/a/dir" "${CURRENT_PACKAGES_DIR}/some/other/dir")


Found 1 error(s). Please correct the portfile:
    /mnt/c/vcpkg/ports/geos/portfile.cmake
-- Performing post-build validation done
Error: Building package geos:x64-dynamic-linux failed with: POST_BUILD_CHECKS_FAILED
Please ensure you're using the latest portfiles with `./vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: geos:x64-dynamic-linux
  Vcpkg version: 2021-01-13-unknownhash
````
Both 
````
/mnt/c:/vcpkg/buildtrees/geos/x64-dynamic-linux-dbg/lib
```` 
and
````
/mnt/vcpkg/buildtrees/geos/x64-dynamic-linux-rel/lib
````
contains: 
````
libgeos_cd.so
libgeos_cd.so.1
libgeos_cd.so.1.16.2
libgeosd.so
libgeosd.so.3.9.0
````
Please advise/fix.
PS
I use the following command line:
````
./vcpkg install aws-sdk-cpp[s3]:x64-dynamic-linux tbb:x64-dynamic-linux hdf5[cpp]:x64-dynamic-linux sqlite3:x64-dynamic-linux tinyxml2:x64-dynamic-linux gtest:x64-dynamic-linux gdal:x64-dynamic-linux log4cxx:x64-dynamic-linux --overlay-triplets=custom-triplets
````

**Comment by dg0yt** [comment]
The empty dirs are `.../bin` not `.../lib`. For some reason, the port file removes them only for static builds. I have another change to this file, so I can look at this issue, too.

**Comment by dg0yt** [comment]
> For some reason, the port file removes them only for static builds.

I assume this was designed for windows, only looking at DLLs. My change is handling `geos-config`, in that directories.

**Comment by JackBoosY** [comment]
Solved by #17616.
